scripts/typescript/src/pppdb/pppdb.ts

```diff
@@ -0,0 +1,257 @@
+   */
+  private createReferenceForMaterial1(data: ChiData) {
+    let result: Required<Pick<IReference, 'node' | 'type' | 'title' | 'doi'>> | undefined;
+
+    //
+    // Strategy:
+    // - Try to get reference first (column AH), then fallback on doi (column A).
+    // - When using reference, it should be 'doi:' or 'isbn:' prefixed, if we throw an exception.
+    // - If both reference and doi are undefined the we throw an exception too.
+    //
+    // Limitations:
+    // - Do not handle semi-colon separated values (ex: chi.xlsx line 109: "DOI: 10.1021/ma50006a079 ; DOI: 10.1016/j.polymer.2007.09.039" )
+    // - references starting with "Citation:" are not handleds
+    //
+
+    if( data.reference ) {
+      const lower_case_doi_or_isbn = data.reference.toLocaleLowerCase();
+
+      if ( lower_case_doi_or_isbn.startsWith('doi:')) {
+        result = {
+          node: ['Reference'],
+          title: '',
+          doi: lower_case_doi_or_isbn.replace('doi:', '').trim(),
+          type: 'journal_article'
+        }
+      } else if (lower_case_doi_or_isbn.startsWith('isbn:')) {
+        result = {
+          node: ['Reference'],
+          title: '',
+          doi: lower_case_doi_or_isbn.replace('isbn:', '').trim(), // TODO: ISBN should NOT be stored as a DOI. We need to identify where we should store it.
+          type: 'book'
+        }
+      } else {
+        this.debug(`Unknown reference: ${data.reference}. Only handle 'doi:' and 'isbn:' for now.`);
+      }
+    } else if ( data.doi ) {
+      result = {
+        node: ['Reference'],
+        title: '',
+        doi: data.doi.toLocaleLowerCase().trim(),
+        type: 'journal_article'
+      }
+    } else {
+      this.debug(`Cannot find any reference or doi. This case is not implemented yet.`)
+    }
+
+    return result;
+  }
+
+  /**
+   * Loads to first sheet of a given *.xslx file
+   * 
+   * User has to know in advance the type of the expected JSON,
+   * No check is done prior to return the data. In case you want to ensure the data has the right type
+   * you must not set T and do discrimination tests.
+   */
+  private loadSheet<T extends { row: number }>(file_path: string, options?: XLSX.ParsingOptions) {
+
+    // Load the XLSX file
+    process.stdout.write(`-- Reading ${file_path} ...\n`);
+    const workBook = XLSX.readFile(file_path, options);
+
+    // Pick the first sheetName
+    const sheetName = workBook.SheetNames.at(0);
+    if( !sheetName ) throw new Error(`Unable to find any sheets in ${file_path}`)
+
+    // Loading the sheet
+    process.stdout.write(`-- Getting '${sheetName}' WorkSheet ...\n`);
+    const workSheet: XLSX.WorkSheet = workBook.Sheets[sheetName] ?? [];
+  
+    // Convert to objects
+    //   Note: We'll loose in performance and memory usage by doing this, but our dataset is very small.
+    //         We prefer here to deal with objects instead of defining a Column enum (cf. rcbc project).
```

Comment:
```suggestion
    //         We prefer here to deal with objects instead of defining a Column enum like we did for rcbc project (the result was not well readable).
```